devtools/x/src/nextest.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    cargo::{
+        build_args::{BuildArgs, Coloring},
+        selected_package::SelectedPackageArgs,
+        CargoCommand,
+    },
+    context::XContext,
+    Result,
+};
+use cargo_metadata::Message;
+use guppy::PackageId;
+use std::ffi::OsString;
+use structopt::StructOpt;
+use testrunner::{
+    reporter::{Color, ReporterOpts, TestReporter},
+    runner::TestRunnerOpts,
+    test_filter::TestFilter,
+    test_list::{TestBinary, TestList},
+};
+
+#[derive(Debug, StructOpt)]
+pub struct Args {
```

Comment:
are you planning to attempt the use of grcov at some point - if we always use parallelism that may prove problematic for to database building.

Reply:
wondering if you've thought about this?

Reply:
I honestly haven't thought about grcov yet (I think getting ignored/doctests and figuring out leaky/failing tests is higher pri), but it seems like it would be possible to do as a next step

Reply:
ignored seems straight forward unless I'm missing something -- adding the flag to all calls would work?.    

I haven't seen doc tests take any substantial amount of time, I suppose they could get compute heavy.   But if cargo treats them as one test I wouldn't try to fight it -- haven't investigated yet.

grcov would probably require running in serial, or finding a way to aggregate the gcda/gcno files, but we could also use lcov.   https://stackoverflow.com/questions/3166198/with-gcov-is-it-possible-to-merge-to-gcda-files.  Perhaps the extra overhead would negate the benefits of running in parallel though?

Reply:
> ignored seems straight forward unless I'm missing something -- adding the flag to all calls would work?

I think that `--ignored` disables non-ignored tests so we need to keep track of the status of every test :(

Reply:
Looks like there's an --include-ignored in nightly:   https://github.com/rust-lang/cargo/issues/7540

Reply:
Yeah, I'd like to try my best not to have dependencies on nightly features though... anyway it should be pretty easy to handle ignored tests, going to do that now.

Reply:
Done in https://github.com/diem/diem-devtools/commit/ccb4a1bc49877d3b354321939a04792cca3e74a0